modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java

```diff
@@ -60,11 +66,97 @@
 
     HashMap<String, HdfsFile> openedFiles;
 
+    private Timer timeReap;
+
+    private Lock lock;
+
+    class TimeReap extends TimerTask {
+        private Map<String, HdfsFile> files;
+        public TimeReap(Map<String, HdfsFile> files) {
+            logger.trace("Create TimeReap");
+            this.files = files;
+        }
+
+        private boolean closeFile(HdfsFile hdfsfile) {
+           logger.info("Close file" + hdfsfile.getPath());
+
+           try {
+               hdfsfile.flush();
```

Comment:
There is no need for the `flush()`, as it does not do much. I wanted to call it, because later on this could mean actual flush. But sure I can remove it now, and we can add if needed.

Reply:
As we speak in IRL, I extend the calls where `close()` is called with `flush()` covering the feature where it makes sense to call `flush()`.